daemon/api.go

```diff
@@ -457,13 +464,41 @@ func getStore(c *Command) snapstate.StoreService {
 	return snapstate.Store(st)
 }
 
+func getSections(c *Command, r *http.Request, user *auth.UserState) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	theStore := getStore(c)
+
+	sections, err := theStore.Sections(user)
+	switch err {
+	case nil:
+		// pass
+	case store.ErrEmptyQuery, store.ErrBadQuery:
+		return BadRequest("%v", err)
+	case store.ErrUnauthenticated:
+		return Unauthorized("%v", err)
+	default:
+		return InternalError("%v", err)
+	}
+
+	meta := &Meta{
+		Sources: []string{"store"},
```

Comment:
Sources is a bit of historical baggage from the ancient API which mixed store and local snaps under a single endpoint. I hope we can even drop that entirely at some point, so it's probably worth keeping this out of this new endpoint.